kamon-core/src/main/scala/kamon/trace/TracingContext.scala

```diff
@@ -66,7 +66,7 @@ private[trace] class TracingContext(traceName: String, token: String, izOpen: Bo
       val segment = currentSegments.next()
       if (segment.isClosed)
         segmentsInfo += segment.createSegmentInfo(_startTimestamp, startTimestamp)
-      else
+      else if (log != null)
```

Comment:
Yeah, I had some NullPointerException when working with simple-trace and some futures : **TracerModule** currently instantiate **TraceContext** with a null **LoggingAdapter** in function [*createTraceContext*](https://github.com/kamon-io/Kamon/blob/master/kamon-core/src/main/scala/kamon/trace/TracerModule.scala#L121).  
I didn't want to change the behavior of **TracerModule** so I added this check to avoid any exception with what I tested, but I think it could throw others errors, for example in the case of a [**MetricsOnlySegment**](https://github.com/kamon-io/Kamon/blob/master/kamon-core/src/main/scala/kamon/trace/MetricsOnlyContext.scala#L105). Maybe the *log* parameter should be an **Option[LoggingAdapter]** if you still agree to instantiate contexts with no logger, to avoid any problems ?

Another thing to note is that [**MetricsOnlySegment**](https://github.com/kamon-io/Kamon/blob/master/kamon-core/src/main/scala/kamon/trace/MetricsOnlyContext.scala#L104) use a check before trying to send a warning where as [**TracingContext**](https://github.com/kamon-io/Kamon/blob/master/kamon-core/src/main/scala/kamon/trace/TracingContext.scala#L69) does not.